### ISSUE TITLE
Feature request embedded audio and video 

### ISSUE THREAD

**Issue by ryanmusante** [Siriusmart/youtube-tui#38]
Is it out of the range of possibilities to have embeded video/audio within the youtube-tui interface rather than externally loading mpv?

**Comment by Siriusmart** [comment]
working on it

**Comment by Siriusmart** [comment]
heres my current progress

`prop` and `sprop` (set prop) are the custom commands i added, but `loadfile` and `stop` are default libmpv commands

now all i need to add is a message bar progress line and now-playing info thing, and the rest can be done using command bindings

https://github.com/Siriusmart/youtube-tui/assets/71584876/6d0b9326-30f5-4cbd-82cd-3c7de4fd77a5

**Comment by ryanmusante** [comment]
Omg thats awesome! I love it

**Comment by Siriusmart** [comment]
https://github.com/Siriusmart/youtube-tui/assets/71584876/aa065177-dbd3-4836-912e-56bcf33c98f4

case closed, embedded video player stuff will not implement, because it already works perfectly

**Comment by Siriusmart** [comment]
so thats latest git, im gonna refine some config and messages (such as the incorrect "player resumed") before pushing to crates.io, oh and i almost forgot to write docs

**Comment by Siriusmart** [comment]
so how it works is that the entire tui rerenders itself every second when a song is playing, as there is no partial rendering feature in the crate im using

this ofc will increase resource usage and stuff, but since the image is only thing that takes resources to render, and it is only rerendered when absolutely nesessary, the performance impact is still near 0

**Comment by Siriusmart** [comment]
also if u see the tui panicking when reaching the end of songs, sometimes, just note that its fixed, not commited yet

the error is on [this line](https://github.com/Siriusmart/youtube-tui/blob/183ac39ccbc258b98e50a430819b2a83c23bc328/src/items/messagebar.rs#L82) and it actually happens because there is an integer overflow when seeker len is not large enough, and `.repeat()` just decided to put a basically infinitely large string into ram

in short, the current latest git is not stable, you'll most likely get interrupted by crashes every few minutes, so dont use it

**Comment by Siriusmart** [comment]
progress update, its done, but still cannot be published to crates.io for reasons
https://github.com/Siriusmart/youtube-tui/releases/tag/v0.8.0

**Comment by ryanmusante** [comment]
The issue with keybindings is gone && your work on embedded is amazing! Great work!

**Comment by ryanmusante** [comment]
It's not just good, it's phenomenal!

**Comment by Siriusmart** [comment]
ty, i took some inspirations from other tuis like [invidtui](https://github.com/darkhz/invidtui), who suggested the same feature on reddit a while back but i didnt feel like doing it

anyways enjoy
